### ISSUE TITLE
AdMob has interstitial but nothing is shown. (After skipping Video Interstitial)

### ISSUE THREAD

**Issue by 0megaD** [sdkbox/issues#32]
In our app we have some code that runs if you have an available interstitial. Usually this code works fine. But sometimes it claims that an ad is available, but nothing shows.

SDKBOX Version: v1.0.1.26
SDKBOX Admob version: v2.3.17.3

```
if(sdkbox::PluginAdMob::isAvailable("interstitial")) {
   sdkbox::PluginAdMob::show("interstitial"); <- shows the ad.
}
```

This code will work most of the time, but sometimes isAvailable will return true, yet show("interstitial") will not trigger any of the callbacks, nor will it show the interstitial.

This happens as soon as the user has seen and skipped a Video Interstitial. (Not the video reward! A normal video interstitial).

So to reproduce:

1.) Create an interstitial adId.
2.) Check if it is available in the code
3.) Show interstitial
4.) If the interstitial is a Video interstitial, press the skip button in the bottom right.
5.) Interstitials will no longer show.

**Comment by yinjimmy** [comment]
checking on this, Thanks for your reporting.

**Comment by yinjimmy** [comment]
> 4.) If the interstitial is a Video interstitial, press the skip button in the bottom right.

It's hard to get a Video interstitial. ;(

**Comment by yinjimmy** [comment]
```
adViewDidFailToReceiveAdWithError:home error message[Request Error: No ad to show.]
```

**Comment by yinjimmy** [comment]
Fixed it. will be released on 12/07 or 08.

**Comment by yinjimmy** [comment]
lets waiting 2 hours for staging build.

**Comment by 0megaD** [comment]
Thanks for the fast response. :)
I will test the update sometime soon when I see it's available.

**Comment by yinjimmy** [comment]
when you skip interstitial video, any callback was trigged?

```
        virtual void adViewDidFailToReceiveAdWithError(const std::string &name, const std::string &msg) {}
        virtual void adViewDidDismissScreen(const std::string &name) {}
        virtual void adViewWillDismissScreen(const std::string &name) {}
```

?

**Comment by 0megaD** [comment]
When I skip interstitial video:
`void MyAdMobListener::adViewWillDismissScreen(const std::string &name)` is called
`void adViewDidFailToReceiveAdWithError(const std::string &name, const std::string &msg)` is not called

Not sure about `adViewDidDismissScreen` (There was an issue in an older version so we use `didDismiss` on android and `willDismiss` on iOS)

**Comment by yinjimmy** [comment]
I think there is a bug in Google iOS SDK. When user skip the video interstitial, the GADInterstitial.isReady is always true.

**Comment by yinjimmy** [comment]
@0megaD , could u plz test with 2.3.17.5 ? It has been released on 20171208.

**Comment by 0megaD** [comment]
I will ask if I can test it tomorrow (need to make a separate admobId at work so that the existing users of our app don't get the bug with the video ad)

I'll update when I have done so.

**Comment by yinjimmy** [comment]
Yes, u can test it. But if you installed 2.3.17.5 before, u need to uninstall or degrade AdMob plugin.

## uninstall
Just remove `.sdkbox_packages.json`

or
## degrade 
edit https://github.com/sdkbox/sdkbox-sample-admob/blob/master/cpp/.sdkbox_packages.json#L29 

Thanks,

**Comment by gl00k** [comment]
I have the same problem
please tell me in version 2.3.17.5 is it resolved?

**Comment by yinjimmy** [comment]
yes, it must be fixed in version 2.3.17.5.
